src/libs/PolicyUtils.ts

```diff
@@ -1205,6 +1206,57 @@ function canModifyPlan(policyID?: string) {
     return !!policy && isPolicyAdmin(policy);
 }
 
+function getAdminsPrivateEmailDomains(policy?: Policy) {
+    if (!policy) {
+        return [];
+    }
+
+    const adminDomains = Object.entries(policy.employeeList ?? {}).reduce((domains, [email, employee]) => {
+        if (employee.role !== CONST.POLICY.ROLE.ADMIN) {
+            return domains;
+        }
+        domains.push(Str.extractEmailDomain(email).toLowerCase());
+        return domains;
+    }, [] as string[]);
+    const ownerDomains = policy.owner ? [Str.extractEmailDomain(policy.owner).toLowerCase()] : [];
+
+    return [...new Set([...adminDomains, ...ownerDomains])].filter((domain) => !isPublicDomain(domain));
```

Comment:
It feels like there are too many spread operators here that aren't really necessary (and spread operators translate to slow code). It feels more like this could be:

```suggestion
    return adminDomains.concat(ownerDomains).filter((domain) => !isPublicDomain(domain));
```

Reply:
I still need to add them all in a Set because I want them to be distinct